kamon-annotation/src/main/scala/kamon/annotation/instrumentation/AnnotationInstrumentation.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * =========================================================================================
+ * Copyright © 2013-2015 the kamon project <http://kamon.io/>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ * =========================================================================================
+ */
+
+package kamon.annotation.instrumentation
+
+import java.lang.reflect.Modifier
+
+import kamon.annotation.util.{ EnhancedELProcessor, ELProcessorPool }
+import org.aspectj.lang.annotation._
+import org.aspectj.lang.{ JoinPoint, ProceedingJoinPoint }
+
+@Aspect
+class AnnotationInstrumentation extends BaseAnnotationInstrumentation {
+
+  @After("execution((@kamon.annotation.Metrics Profiled+).new(..)) && this(profiled)")
+  def creation(profiled: Profiled): Unit = {
+
+    import EnhancedELProcessor.Syntax
+
+    implicit val stringEvaluator = StringEvaluator { str ⇒ ELProcessorPool.useWithObject(profiled)(_.evalToString(str)) }
+
+    profiled.getClass.getDeclaredMethods.filterNot(method ⇒ Modifier.isStatic(method.getModifiers)).foreach {
+      method ⇒
+        registerTrace(method, profiled.traces)
+        registerSegment(method, profiled.segments)
+        registerCounter(method, profiled.counters)
+        registerMinMaxCounter(method, profiled.minMaxCounters)
+        registerHistogram(method, profiled.histograms)
+        registerTime(method, profiled.histograms)
+    }
```

Comment:
I take the absolute method name, for example: `public-static-long-kamon.annotation.annotatedjavaclass.histogram(java.lang.long)` then an overloaded version could be  `public-static-long-kamon.annotation.annotatedjavaclass.histogram(java.lang.long, java.lang.String)` or `public-static-double-kamon.annotation.annotatedjavaclass.histogram(java.lang.long, java.lang.String)` this is done inside of register* methods.